Refactor + fix request sender used by Alertmanager notifier

rwwiv opened the pull request:
This PR does two things: 
- It moves away from using an exported function defined as a variable in `util.go` (i.e. sendHTTPRequest) in favor of an interface (Sender) with the intention that this be the first step towards pulling this code out of the library. 
- It also fixes this [issue](https://github.com/grafana/grafana/issues/84859) by ensuring the short-lived client created when sending alerts to an Alertmanager receiver does not keep the created connections alive.

<!-- CURSOR_SUMMARY -->
---

> [!NOTE]
> **Medium Risk**
> Touches the outbound HTTP delivery path for Alertmanager notifications; while behavior should be equivalent, transport-level changes (keep-alives disabled, new sender abstraction) could affect performance or edge-case connectivity.
> 
> **Overview**
> Refactors Alertmanager notification delivery to use an injected `receivers.Sender` (`Notifier.sender`) instead of the package-level, stubbable `receivers.SendHTTPRequest` variable.
> 
> Moves the HTTP POST implementation into `defaultSender.SendHTTPRequest`, and **disables HTTP keep-alives** for this short-lived client to avoid lingering connections. Tests are updated to mock via the new interface and add coverage (`TestSenderDisablesKeepAlives`) ensuring connections are not retained.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit ea215f7753ee1f79b1dd99424220eef81c16bf5f. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>
<!-- /CURSOR_SUMMARY -->

rwwiv committed:
Refactor alertmanager sendHttpRequest to interface

Disable keep alives and add test to verify

Fix lint

rwwiv commented on receivers/util.go:
```diff
@@ -212,3 +154,71 @@ func TruncateInBytes(s string, n int) (string, bool) {
+			Timeout: 30 * time.Second,
+		}).DialContext,
+		TLSHandshakeTimeout: 5 * time.Second,
+		// Disable keep alive since this is a short lived client
+		DisableKeepAlives: true,
```
This is the line that fixes the bug mentioned in the PR desc.

wiemerc commented:
Is this PR still being worked on? We would really appreciate if it least the fix could be merged because we are affected by this issue too.

benmali commented:
We are facing a similar issue, any chance to merge this?

rwwiv committed:
Merge branch 'main' into rwwiv/am-receiver-sender-cleanup

# Conflicts:
#	receivers/util.go

rwwiv committed:
Fix logging format and test issues from PR review

- Add missing 'msg' key to log statements in util.go to follow go-kit/log
  key-value pair convention
- Use context-enriched logger in alertmanager.go for consistent logging
- Update alertmanager tests to use mockSender pattern instead of removed
  global SendHTTPRequest function
- Fix util_test.go to use log.NewNopLogger() instead of non-existent
  logging.FakeLogger

moustafab reviewed (approved):
LGTM, the cursor comment is interesting though might want to consider that

rwwiv committed:
Merge branch 'main' into rwwiv/am-receiver-sender-cleanup
